ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/requests/RequestUtil.java

```diff
@@ -40,6 +42,23 @@ public static <T extends Request> List<T> filterRequestsOfType(
     return requests.stream().filter(requestType::isInstance).map(requestType::cast).toList();
   }
 
+  public static Optional<List<Deposit>> getDepositRequests(final Optional<List<Request>> requests) {
+    return requests.map(r -> filterRequestsOfType(r, Deposit.class));
+  }
+
+  public static Optional<List<WithdrawalRequest>> getWithdrawalRequests(
+      final Optional<List<Request>> requests) {
+    return requests.map(r -> filterRequestsOfType(r, WithdrawalRequest.class));
+  }
+
+  /**
+   * Combines two optional lists of requests into a single optional list.
+   *
+   * @param maybeDeposits Optional list of deposit requests.
```

Comment:
nit - for consistency should this be renamed to maybeDepositsRequest

Reply:
actually on further consideration I see that matches the class names Deposit and WithdrawalRequest

Reply:
We have Withdrawal and WithdrawalRequest but Deposit is itself the request... it is a bit confusing but probably outside the scope of this PR. non-blocking.

Reply:
I will rename Deposit to DepositRequest but I wanted to do that in a different PR so it is easier to review it.